src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -468,7 +480,7 @@ func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
 			"--overprovisioned",
 			// sometimes a little bit of memory is consumed by other processes than seastar
 			"--reserve-memory " + redpandav1alpha1.ReserveMemoryString,
-			"--smp=1",
+			"--smp=" + strconv.FormatInt(smp, 10),
```

Comment:
should we make sure is at least 1?

Reply:
In the updated version when in developer mode, smp is either positive or not passed as an argument.